moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -304,16 +293,31 @@ bool RuckigSmoothing::runRuckig(robot_trajectory::RobotTrajectory& trajectory,
       getNextRuckigInput(trajectory.getWayPointPtr(waypoint_idx), next_waypoint, group, ruckig_input);
 
       // Run Ruckig
-      ruckig_result = ruckig_ptr->update(ruckig_input, ruckig_output);
+      ruckig_result = ruckig.update(ruckig_input, ruckig_output);
+
+      // The difference between Result::Working and Result::Finished is that Finished can be reached in one
+      // Ruckig timestep (constructor parameter). Both are acceptable for trajectories.
+      // (The difference is only relevant for streaming mode.)
 
-      if ((waypoint_idx == num_waypoints - 2) && ruckig_result == ruckig::Result::Finished)
+      // If successful and at the last trajectory segment
+      if ((waypoint_idx == num_waypoints - 2) &&
+          (ruckig_result == ruckig::Result::Working || ruckig_result == ruckig::Result::Finished))
```

Comment:
wouldn't this lead to trajectories being aborted early? Maybe I'm reading it wrong. I see why this would be a good mode for streamed commands though.

Reply:
No, Result::Working means Ruckig did find a solution but it wasn't within the timestep that Ruckig was initialized with at [L277](https://github.com/ros-planning/moveit/blob/7bb2d2961fa092de3a6039c0624d156b167e0a62/moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp#L277).

That's acceptable because we set the correct waypoint duration at L306.

BTW I wouldn't mind checking with @pantor to see if we can get some paid help with optimizing this at some point. I know there are some further improvements that can be made.